scripts/dts/gen_defines.py

```diff
@@ -650,7 +650,11 @@ def write_vanilla_props(node):
             # DT_N_<node-id>_P_<prop-id>_IDX_<i>_EXISTS
             for i, subval in enumerate(prop.val):
                 if isinstance(subval, str):
+                    subprop = edtlib.Property(None, subval, None)
+
                     macro2val[macro + f"_IDX_{i}"] = quote_str(subval)
+                    macro2val[macro + f"_IDX_{i}_STRING_TOKEN"] = subprop.val_as_token
+                    macro2val[macro + f"_IDX_{i}_STRING_UPPER_TOKEN"] = subprop.val_as_token.upper()
```

Comment:
BTW, you can use str_as_token() now instead of creating a fake property. https://github.com/zephyrproject-rtos/zephyr/commit/4ef8c8e44711f68798ff51b8b283003c68e31fcd

Reply:
Please note that there was some change provided without any tests and documentation. I am adding my below, but consider removing the previous and left the _STRING_TOKEN version as provided with the whole process, with the documentation and macros.